internal/stacksapi/stack_notification_test.go

```diff
@@ -32,53 +40,77 @@ func TestCreateStackNotification(t *testing.T) {
 
 			w.Header().Set("X-Custom-Header", "custom-value")
 			w.WriteHeader(http.StatusOK)
+			assert.NoError(t, json.NewEncoder(w).Encode(CreateStackNotificationsResponse{
+				Errors: []StackNotificationError{},
+			}))
```

Comment:
Going forward, I would greatly appreciate it if test errors identified the function called and any relevant args, if not too long. `assert.NoError` doesn't log this information on its own.

```suggestion
			err := json.NewEncoder(w).Encode(CreateStackNotificationsResponse{
				Errors: []StackNotificationError{},
			})
			if err != nil {
				t.Errorf("json.NewEncoder(w).Encode(CreateStackNotificationsResponse{...}) error = %v", err)
			}
```

Reply:
Are you saying that we shouldn't use assert.NoError? I think there is a trade-off to be made here. 

Why I think assert.NoError is good: 

1. Many errors in test are so rare that having that log line is more harm than good, as it reduce code readibiliy. 
2. Even with the log line, in most case I believe people will need go back to the test file to understand the context. 

So I think there is an imbalance on the benefit vs cost here. Do you feel strongly about it? Maybe we can talk in sync, if we can get a consensus then maybe we can make a linter rule?

Reply:
If it were up to me, `assert` would be banned from existence. But I realise that is an extreme opinion. 

A good error log message practically eliminates the need to re-read the test in order to understand the failure. Indenting the error flow makes it easier to "skip over" when reading. I think `assert.NoError` is _harder_ to read, because it doesn't look like the common error handling pattern in Go (`if err != nil {...}`).

Reply:
Maybe the issue here is that I use `assert.NoError` on the same line with the calling code? 

If I split them, would this be more readable? It resembles the canonical error flow. There are a few cases in codebase, which we haven't received readability complaint in previous reviews. 

```golang
err := json.NewEncoder(w).Encode(...)
assert.NoError(t, err) 
```

----

On the other hand, if we want to ban it, I will be in favor of an automatic approach, but be against manual enforcement.